src/quali_can.c

```diff
@@ -126,20 +156,74 @@ static void can_transmit_task(void *arg)
 
     while (!handle_priv->handle_data.stop_test) {
         esp_err_t ret;
+        char msg[MAX_MSG_SIZE];
         ret = twai_transmit(&message, portMAX_DELAY);
         if (ret != ESP_OK) {
             ESP_LOGI(TAG, "Transmit Error: ret = 0x%x", ret);
-            sprintf(report, "ERR: Transmit Error: ret = 0x%x\n", ret);
-            reporter->report_status(reporter, report);
+            snprintf(msg, MAX_MSG_SIZE, "ERR: Transmit Error: ret = 0x%x\n", ret);
+            reporter->report_status(reporter, msg);
             vTaskDelay(pdMS_TO_TICKS(500));  // don't flood host with errors
         } else {
-            handle_priv->handle_data.msg_counter++;
-            handle_priv->handle_data.total_data += 8;
+            handle_priv->handle_data.tx_msg_counter++;
+            handle_priv->handle_data.total_tx_data += 8;
         }
     }
 
+    // notify receive task that driver is ready to stop
+    xSemaphoreGive(handle_priv->handle_data.driver_stop);
+
+    vTaskDelete(NULL);
+}
+
+static void can_receive_task(void *arg)
```

Comment:
due to the receive task CAN alters check is no longer necessary. remove CAN alerts